extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -81,4 +103,80 @@ private void addReflectiveHierarchyClass(DotName className) {
     private void addReflectiveClass(boolean methods, boolean fields, String... className) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, className));
     }
+
+    // Generate a ObjectMapperCustomizer bean that registers each serializer / deserializer with ObjectMapper
+    @BuildStep
+    void generateCustomizer(BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            List<JacksonModuleBuildItem> jacksonModules) {
+
+        if (jacksonModules.isEmpty()) {
+            return;
+        }
+
+        ClassOutput classOutput = new ClassOutput() {
+            @Override
+            public void write(String name, byte[] data) {
+                generatedBeans.produce(new GeneratedBeanBuildItem(name, data));
+            }
+        };
+
+        try (ClassCreator classCreator = ClassCreator.builder().classOutput(classOutput)
+                .className("io.quarkus.jackson.customizer.RegisterSerializersAndDeserializersCustomizer")
+                .interfaces(ObjectMapperCustomizer.class.getName())
```

Comment:
Why do you use full String for `className` and the class for `interfaces` (in which you call `getName`).
Maybe use String for both will be more concistent.
As I understood, it is favored in build steps to avoid unecessary class loading.

Reply:
Good question. In this setting (strictly inside a build step) using the class doesn't cause problems. Moving classes around via Build Items can however.
This sort of pattern is used throughout Quarkus

Reply:
Thanks for the explaintation. A new day, a new learning :)

Reply:
Glad to help anytime :)